caddyhttp/httpserver/context.go

```diff
@@ -26,6 +26,7 @@ import (
 // Context is the context with which Caddy templates are executed.
 type Context struct {
 	Root http.FileSystem
+	Res  http.ResponseWriter
```

Comment:
This gets exposed to the templates, so just with this line, templates can now do:

```
{{.Res.Header.Add "Field" "Value"}}
```

I can see why having a convenience method for push would be useful though:

```
{{.Push "/filename.ext"}}
```